src/gsi/crtm_interface.f90

```diff
@@ -2597,6 +2693,185 @@ subroutine calc_gfdl_reff(rho_air,tsen,qxmr,cloud_name,reff)
 
   end subroutine calc_gfdl_reff
 
+  subroutine calc_thompson_reff(rho_air,tsen,qxmr,cloud_name,reff)
+!$$$  subprogram documentation block
+!                .      .    .                                       .
+! subprogram:    calc_thompson_reff:  calculate effective radius based on Thompson scheme
+!   prgmmr:      Azadeh.Gholoubi
+!
+! program history log:
+!
+!   2023-01-24
+!
+!   language: f90
+!
+!$$$
+!--------
+  use constants, only: zero, pi
+  implicit none
+
+! Declare passed variables
+  character(10)                 ,intent(in   ) :: cloud_name
+  real(r_kind), dimension(nsig) ,intent(in   ) :: rho_air   ! [ kg/m3  ]
+  real(r_kind), dimension(nsig) ,intent(in   ) :: tsen      ! [ K      ]
+  real(r_kind), dimension(nsig) ,intent(in   ) :: qxmr      ! [ kg/kg  ]
+  real(r_kind), dimension(nsig) ,intent(inout) :: reff      ! [ micron ]
+
+! Declare local variables
+  character(len=*), parameter :: myname_ = 'calc_thompson_reff'
+  integer(i_kind) :: k
+  integer(i_kind) :: j
+  real(r_kind)    :: qx
+  real(r_kind)    :: reff_min, reff_max
+  real(r_kind):: lam_i,lam_w,lam_s,lam_r, lam_g,lam_exp, am_r,am_w,am_i,am_g, N_0r ,N_0i,N_0w,mu_w
+
+  ! Parameters
+  real(r_kind), parameter :: qmin = 1.0e-12_r_kind          ! [kg/kg ]
+  !.. droplet number concentration.
+  real, parameter :: Nt_c = 100.E6  ![m-3]
+
+  ! Parameters for water cloud
+  real(r_kind), parameter :: ccn        =  1.0e8_r_kind
+  real(r_kind), parameter :: rho_w      = 1000.0_r_kind     ! [kg/m3 ]
+  real(r_kind), parameter :: reff_w_min =    2.0_r_kind     ! previous value was 5.0_r_kind
+  real(r_kind), parameter :: reff_w_max =   25.0_r_kind     ! previous value was 10.0_r_kind
+
+  ! Parameters for ice cloud (Hemisfield and mcFarquhar 1996)
+  real(r_kind), parameter :: rho_i      = 890.0_r_kind      ! [kg/m3 ]
+  real(r_kind), parameter :: reff_i_min =    2.5_r_kind     ! previous value was 10_r_kind
+  real(r_kind), parameter :: reff_i_max = 250.0_r_kind      ! previous value was 150_r_kind
+  real(r_kind), parameter:: mu_i = 0.0
+
+  ! Parameters for  rain (Lin 1983)
+  real(r_kind), parameter :: rho_r      =    1000.0_r_kind  ! [kg/m3 ]
+  real(r_kind), parameter :: reff_r_min =       50.0_r_kind  ! [micron] ! previous value was 0.0_r_kind
+  real(r_kind), parameter :: reff_r_max =   1000.0_r_kind  ! [micron] ! previous value was 10000.0_r_kind
+  real(r_kind), parameter:: mu_r = 0.0
+  ! Parameters for snow
+  real(r_kind), parameter :: reff_s_min =       5.0_r_kind  ! [micron] ! previous value was 0.0_r_kind
+  real(r_kind), parameter :: reff_s_max =   5000.0_r_kind  ! [micron] ! previous value was 10000.0_r_kind
+
+!For snow moments conversions  (from Field et al. 2005)
+  real, dimension(10), parameter:: &
+      sa = (/ 5.065339, -0.062659, -3.032362, 0.029469, -0.000285,      &
```

Comment:
Given the specified precision of `sa`, append this precision to each parameter value.